test/test_heuristics.rb

```diff
@@ -13,8 +13,9 @@ def file_blob(name)
   end
 
   def all_fixtures(language_name, file="*")
-    Dir.glob("#{samples_path}/#{language_name}/#{file}") -
-      ["#{samples_path}/#{language_name}/filenames"]
+    fixs = Dir.glob("#{samples_path}/#{language_name}/#{file}") -
+           ["#{samples_path}/#{language_name}/filenames"]
+    fixs.reject { |f| File.symlink?(f) }
```

Comment:
Do we need to filter out the symlinks during testing?  Wouldn't doing so effectively mean we never test a symlink with the heuristics?

Additionally, I think we need a specific heuristics test for symlinks to ensure the heuristics are being skipped for them.

Reply:
The problem is the test symlink's placement in the `Markdown` directory means we assume it should return Markdown as the result, whereas we skip heuristics instead and return nothing.  Before this change to the test, we get this output:

``` console
  1) Failure:
TestHeuristics#test_md_by_heuristics [/Users/kivikakk/github/linguist/test/test_heuristics.rb:191]:
Failed for /Users/kivikakk/github/linguist/samples/Markdown/symlink.md.
--- expected
+++ actual
@@ -1 +1 @@
-[#<Linguist::Language name=Markdown>]
+[]
```

I'll add an explicit test too.